system_tests/resources/linux/scripts/test_net.py

```diff
@@ -44,13 +46,55 @@ def get_interface(mac):
     return interface
 
 
-def set_ip_address(interface, ip):
-    run(
-        'ip addr add {ip}/24 dev {interface}'.format(
-            ip=ip,
+def interface_has_ip(interface, ip):
+    with settings(
+        hide('warnings', 'running', 'stdout', 'stderr'),
+        warn_only=True,
+    ):
+        command = 'ip addr show {interface} | grep {ip}'.format(
             interface=interface,
+            ip=ip,
         )
-    )
+        result = run(command)
+    if result.return_code == 0:
+        return False
+    elif result.return_code == 1:
+        return True
+
+
+def set_ip_address(interface, ip):
+    for i in range(0, 30):
+        if i == 0:
```

Comment:
I felt like doing that would also mean that the logic for applying the ip should be pulled out (probably into its own method) and put both with the initial logging line and at the end of the loop, which made it feel like it would look clumsier to me.

I might be too close to the coal face on this one- which way do you think would read better?